packages/ERTP/src/issuer.chainmail

```diff
@@ -0,0 +1,209 @@
+  /** Get the Issuer for this mint. */
+  getIssuer() -> (Issuer);
+
+  /**
+   * Create a new Payment containing newly minted amount. 
+   */
+  mintPayment(newAmount :Amount) -> (Payment);
+}
+
+/**
+ * Purses hold amount of digital assets of the same brand, but unlike Payments, they are
+ * not meant to be sent to others. To transfer digital assets, a
+ * Payment should be withdrawn from a Purse. A purse's balance can rise
+ * and fall, through the action of deposit() and withdraw().
+ *
+ * The primary use for Purses and Payments is for currency-like and goods-like
+ * digital assets, but they can also be used to represent other kinds of rights, such
+ * as the right to participate in a particular contract.
+ */
+interface Purse (Amount) {
+
+  /** Get the Issuer for this mint. */
+  getIssuer() -> (Issuer);
+
+  /** Get the amount contained in this purse, confirmed by the issuer. */
+  getBalance() -> (Amount);
+
+  /**
+   * Deposit all the contents of srcPayment into this purse, returning the
+   * amount. If the optional argument amount does not equal the balance of srcPayment,
+   * throw error.
+   */
+  deposit(srcPayment :Payment, amount :Amount) -> (Amount);
+
+  /** Withdraw amount from this purse into a new Payment. */
+  withdraw(amount :Amount) -> (Payment);
+
+}
+
+/**
+ * Payments hold amount of digital assets of the same brand in transit. Payments can
+ * be deposited in purses, split into multiple payments, combined, and
+ * claimed (getting an exclusive payment). Payments are linear, meaning
+ * that either a payment has its full original balance, or it is used up
+ * entirely. It is impossible to partially use a payment. 
+ *
+ * Payments are often received from other actors and therefore should
+ * not be trusted themselves. To get the balance of a payment, use the
+ * trusted issuer: issuer.getBalance(payment),
+ *
+ * Payments can be converted to Purses by getting a trusted issuer and
+ * calling `issuer.makeEmptyPurse()` to create a purse, then `purse.deposit(payment)`.
+ */
+interface Payment (Amount) {
+
+  /** 
+   * Get the allegedBrand, indicating the kind of digital asset this
+   * payment purports to be, and which issuer to use. Because payments 
+   * are not trusted, any method calls on payments should be treated
+   * with suspicion and verified elsewhere.
+   */ 
+  allegedBrand() -> (Brand);
+}
+
+/** 
+ * All of the difference in how digital asset amount are manipulated can be reduced to
+ * the behavior of the math on extents. We extract this
+ * custom logic into mathHelpers. MathHelpers are about extent
+ * arithmetic, whereas AmountMath is about amounts, which are the
+ * extents labeled with a brand. AmountMath use mathHelpers to do their extent arithmetic,
+ * and then brand the results, making a new amount. 
+ */ 
+interface mathHelpers () {
+  /** 
+   * Check the kind of this extent and throw if it is not the
+   * expected kind. 
+   */
+  doAssertKind(allegedExtent :Extent) -> (Extent);
```

Comment:
`doAssertKind` does not return anything in general. It is a validator; it indicates approval by returning normally.

On natMathHelpers, `doAssertKind` does happen to return the approved extent. But not on `strMathHelpers` or `strSetMathHelpers`.

As you shift to having `doAssertKind` do all validation, and the other methods assuming their arguments already validated, you should say so here (or wherever `mathHelpers` moves to).

As you shift towards harmonizing the mathHelpers names with the amountMath names (e.g., empty), we should consider replacing `doAssertKind` with `doCoerce` and have its responsibility changed to a genuine coercer consistent with the declaration here. It is only called once within amountMath's `make`, whose responsibility is already consistent with coercing. 

```js
helpers.doAssertKind(allegedExtent);
```
in amountMath could be
```js
const extent = helpers.doCoerce(allegedExtent);
```
if we want to emphasize the point.

Reply:
I thought we said we weren't doing that because of `Nat` not being a coercer. Is that still right?

Reply:
I did agree with that, but I like the other way more. `Nat`, because it does return its approval, also satisfies the coercer protocol. It is a degenerate coercer. I see only pros and no cons to replace `doAssertKind` with one that return an approved value, which may be the argument or may be closely derived from the argument. Given that it is *only* used by `amountMath.coerce`, I think the parallelism is an additional pro.

Reply:
And of course, the above chainmail declaration would already be correct, aside from the name ;)

Reply:
Done!